<!DOCTYPE html>
<html>

<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Metagenomics Pipeline</title>
  <link rel="stylesheet" href="https://stackedit.io/style.css" />
	<style>
    section.footnotes::before { 
      content: "References";
      margin: 1.8em 0;
      line-height: 1.33;
      display: block;
      font-size: 1.5em;
      margin-block-start: 0.83em;
      margin-block-end: 0.83em;
      margin-inline-start: 0px;
      margin-inline-end: 0px;
      font-weight: bold;
      border-bottom: 1px solid hsla(0,0%,50%,.33);
      position: relative;
      top: .33em;
	}
	hr.footnotes-sep {
	display:none;
	}
  </style>
	<style>
	code {
  -webkit-user-select: all;
  user-select: all;
  display: block;
  padding: 20px;
  border: 2px dashed #c9baa7;
  color: #31808c;
  background-color: #f5f4ef;
  text-align: left;
  font: 0.85rem/1.5 'Roboto Mono', monospace;
  overflow: auto;
}

code:focus {
  border: 2px dashed #31808c;
  outline: none;
  animation: select 100ms step-end forwards;
}

@keyframes select {
  to {
    -webkit-user-select: text;
    user-select: text;
  }
}
</style>

</head>

<body class="stackedit">
  <div class="stackedit__html"><h1 id="megan6-pipeline">MEGAN6 pipeline</h1>
<p>Hello there, here we are at the 4th part of the pipeline, setting up the <a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004957">MEGAN6 Community Edition</a><sup class="footnote-ref"><a href="#fn1" id="fnref1">1</a></sup> pipeline to process the data coming from the <a href="https://uni-tuebingen.de/fakultaeten/mathematisch-naturwissenschaftliche-fakultaet/fachbereiche/informatik/lehrstuehle/algorithms-in-bioinformatics/software/diamond/">DIAMOND</a><sup class="footnote-ref"><a href="#fn2" id="fnref2">2</a></sup> reads aligner.</p>
<ol>
<li>HISAT2</li>
<li>Samtools</li>
<li>DIAMOND</li>
<li><strong>MEGAN6</strong></li>
<li>Plots</li>
</ol>
<div class="mermaid"><svg xmlns="http://www.w3.org/2000/svg" id="mermaid-svg-oUsObZAfXb5E5S9v" width="100%" style="max-width: 921.09375px;" viewBox="0 0 921.09375 191"><g transform="translate(-12, -12)"><g class="output"><g class="clusters"><g class="cluster" id="subGraph0" transform="translate(624.2734375,78)" style="opacity: 1;"><rect width="391.765625" height="116" x="-195.8828125" y="-58"></rect><g class="label"><g transform="translate(0,0)"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"></div></foreignObject></g></g><text x="0" y="-44" fill="black" stroke="none" id="mermaid-svg-oUsObZAfXb5E5S9vText" style="text-anchor: middle;">MEGAN6</text></g></g><g class="edgePaths"><g class="edgePath" style="opacity: 1;"><path class="path" d="M94.734375,128.62470249279718L119.734375,117L144.734375,117" marker-end="url(#arrowhead2242)" style="fill:none"></path><defs><marker id="arrowhead2242" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M228.859375,117L253.859375,117L278.859375,117" marker-end="url(#arrowhead2243)" style="fill:none"></path><defs><marker id="arrowhead2243" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M372.7175480769231,94L403.390625,78L428.390625,78L453.390625,78" marker-end="url(#arrowhead2244)" style="fill:none"></path><defs><marker id="arrowhead2244" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M372.7175480769231,140L403.390625,156L428.390625,156L514.796875,156L601.203125,156L710.6796875,156L820.15625,156L845.15625,156L870.15625,156" marker-end="url(#arrowhead2245)" style="fill:none"></path><defs><marker id="arrowhead2245" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M576.203125,78L601.203125,78L626.203125,78" marker-end="url(#arrowhead2246)" style="fill:none"></path><defs><marker id="arrowhead2246" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M86.64158163265306,169L119.734375,195L186.796875,195L253.859375,195L328.625,195L403.390625,195L428.390625,195L514.796875,195L601.203125,195L710.6796875,195L820.15625,195L845.15625,195L870.15625,176.41751042287075" marker-end="url(#arrowhead2247)" style="fill:none"></path><defs><marker id="arrowhead2247" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M795.15625,78L820.15625,78L845.15625,78L882.1534455128206,133" marker-end="url(#arrowhead2248)" style="fill:none"></path><defs><marker id="arrowhead2248" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g></g><g class="edgeLabels"><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g></g><g class="nodes"><g class="node" id="E" transform="translate(514.796875,78)" style="opacity: 1;"><rect rx="0" ry="0" x="-61.40625" y="-23" width="122.8125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-51.40625,-13)"><foreignObject width="102.8125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">daa-meganizer</div></foreignObject></g></g></g><g class="node" id="F" transform="translate(710.6796875,78)" style="opacity: 1;"><rect rx="0" ry="0" x="-84.4765625" y="-23" width="168.953125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-74.4765625,-13)"><foreignObject width="148.953125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">compute-comparison</div></foreignObject></g></g></g><g class="node" id="A" transform="translate(57.3671875,146)" style="opacity: 1;"><rect rx="0" ry="0" x="-37.3671875" y="-23" width="74.734375" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-27.3671875,-13)"><foreignObject width="54.734375" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">HISAT2</div></foreignObject></g></g></g><g class="node" id="B" transform="translate(186.796875,117)" style="opacity: 1;"><rect rx="0" ry="0" x="-42.0625" y="-23" width="84.125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-32.0625,-13)"><foreignObject width="64.125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">Samtools</div></foreignObject></g></g></g><g class="node" id="C" transform="translate(328.625,117)" style="opacity: 1;"><rect rx="0" ry="0" x="-49.765625" y="-23" width="99.53125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-39.765625,-13)"><foreignObject width="79.53125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">DIAMOND</div></foreignObject></g></g></g><g class="node" id="G" transform="translate(897.625,156)" style="opacity: 1;"><rect rx="0" ry="0" x="-27.46875" y="-23" width="54.9375" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-17.46875,-13)"><foreignObject width="34.9375" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">Plots</div></foreignObject></g></g></g></g></g></g></svg></div>
<h2 id="requirements">Requirements</h2>
<p>MEGAN6 Community Edition 6.16 or greater <a href="https://software-ab.informatik.uni-tuebingen.de/download/megan6/welcome.html">download</a><br>
eggNOG, InterPro2GO and SEED file maps <a href="https://software-ab.informatik.uni-tuebingen.de/download/megan6/welcome.html">download</a><br>
Java runtime environment version 11 or greater <a href="https://www.oracle.com/java/technologies/javase/jdk11-archive-downloads.html">download</a><br>
xvfb-run is needed to run the compute-comparison tool <a href="https://en.wikipedia.org/wiki/Xvfb">Wikipedia</a></p>
<h2 id="pipeline">Pipeline</h2>
<p>The MEGAN6 pipeline we set up is executed using command-line and has two parts:</p>
<ol>
<li><strong>Meganize</strong> all <em>.daa</em> files, using the <em>daa-meganizer</em> tool provided by the MEGAN6, so they can be ready for the second part of the pipeline.</li>
<li><strong>Compute</strong> the comparisons between all “meganized” files using the <em>compute-comparison</em> tool also provided by the MEGAN6.</li>
</ol>
<blockquote>
<p>Note: These tools are inside the folder tools of the MEGAN6 installation.</p>
</blockquote>
<h2 id="explaining-the-pipeline">Explaining the pipeline</h2>
<p>Firstly, the command-line is used to improve the throughput of the pipeline, in this way we can use the supercomputers with multiple-cores available for our research. The first part of the MEGAN6 pipeline, prepares the samples as .daa files in a way they can be read by the <em>compute-comparison</em> tool. The second part of the pipeline, compute all the samples and generates this computation containing all the maps to the databases provided (eggNOG, InterPro2GO and SEED), finally this file, can be analysed inside the MEGAN6 GUI.</p>
<h2 id="pipeline-process">Pipeline process</h2>
<p>We had two super computers available for our computation: NPAD, and Bioinformatics 3. The first one, NPAD has an older version of the MEGAN6 (version 6.12.3, built 14 Aug 2018), that works just fine to <em>meganize</em> using separate database files, but there is a couple drawbacks, it does not contain the compute-comparison tool, and at the date I’m writing this tutorial, it does not have the latest databases available, whereas in the Bioinformatics3 server it was possible to set up with the latest version of the MEGAN6 (version 6.18.5, built 14 Feb 2020) at that time, this version uses only one main database file containing all the updated databases we map our files to. This small difference (one file with all databases, in opposition to separate files) will lead to a different syntax to use the daa-meganizer.</p>
<h3 id="part-1---meganizing-files">Part 1 - Meganizing files</h3>
<h4 id="bioinformatics3-server">Bioinformatics3 server</h4>
<p>Assuming you have downloaded and installed MEGAN6, and also downloaded the database map file, it is a good practice to set up a batch script to process the computation:</p>
<pre><code>#!/bin/bash

#setting constants
global_dir="/data/home/leozenon"
db_dir="/data/home/leozenon/db"
map="/data/home/leozenon/db/megan-map-Oct2019.db"
file="1.INF.daa"

# I have tools as symbolic link from MEGAN tools on my home folder
# 28 Mar  3 18:31 tools -&gt; /data/home/root/megan6/tools

#meganizing
./tools/daa-meganizer -i $db_dir/$file -mdb $map -pr

#moving to another directory after job done
mv $db_dir/$file $db_dir/mega/
</code></pre>
<h4 id="npad-server">NPAD server</h4>
<p>The NPAD server has a different way of setting up it’s jobs that can be read on it’s on FAQ website, firstly, I have copied the daa-meganizer tool to my global folder named as daa-meg, and edited in a way to use more of the server resources:</p>
<pre><code>#!/bin/bash

# Runs the DAA-Meganizer command-line program

#

# Copyright (C) 2019 Daniel H. Huson

#

# Use only as permitted under MEGAN Ultimate Edition license agreement, do not redistribute.

  
  

bin_dir=`dirname "/opt/npad/shared/softwares/python/3.6-anaconda-5.0.1/envs/megan6/opt/megan-6.12.3/tools/daa-meganizer"`

bin_dir=`cd "$bin_dir"  &amp;&amp; pwd` # ensure absolute path

jars_dir="$bin_dir/../jars"

  

jre_dir=/opt/npad/shared/softwares/python/3.6-anaconda-5.0.1/envs/megan6

  

if [ -z $jre_dir ]

then

java=java

vmOptions="-Xmx64G"

classpath="../antbuild/MEGAN.jar:$jars_dir/MALT.jar:$jars_dir/data.jar:"

else

java=$jre_dir/bin/java

vmOptions="-Xmx64000M"

classpath="$jars_dir/MEGAN.jar:$jars_dir/MALT.jar:$jars_dir/data.jar:"

fi

  

options=$*

if [ $# ==  0 ]

then

options="-h"

fi

  

java_flags="-server -Duser.language=en -Duser.region=US -Djava.awt.headless=true $vmOptions"

  

$java $java_flags -cp "$classpath" megan.tools.DAAMeganizer $options
</code></pre>
<p>After that to use the NPAD supercomputer resources (memory and cores) there is an specific syntax which is explained on it’s own <a href="http://npad.ufrn.br/tutoriais/index.php">tutorial</a><sup class="footnote-ref"><a href="#fn3" id="fnref3">3</a></sup>, but I have added the batch script it’s used on this specific pipeline:</p>
<pre><code>#!/bin/bash

  

#SBATCH --time=1-0:0

#SBATCH --mem=64000

  

global_dir="/home/lztassi/global"

db_dir="/home/lztassi/global/db"

acc_nucl="/home/lztassi/global/acc/nucl_acc2tax-Jul2019.abin"

acc_egg="/home/lztassi/global/acc/acc2eggnog-Jul2019X.abin"

acc_inter="/home/lztassi/global/acc/acc2interpro-Jul2019X.abin"

acc_seed="/home/lztassi/global/acc/acc2seed-May2015XX.abin"

file="1.INF.daa"

  

./daa-meg -i $db_dir/$file -pr -alg weighted -a2t $acc_nucl -a2eggnog $acc_egg -a2interpro2go $acc_inter -a2seed $acc_seed

mv $db_dir/$file $db_dir/mega/
</code></pre>
<p>The expected output will be something similar to this:</p>
<pre><code>Version   MEGAN Community Edition (version 6.18.5, built 14 Feb 2020)
Copyright (C) 2019 Daniel H. Huson. This program comes with ABSOLUTELY NO WARRANTY.
Java version: 11.0.1
Functional classifications to use: EGGNOG, INTERPRO2GO, SEED
Loading ncbi.map: 2,175,506
Loading ncbi.tre: 2,175,510
Loading eggnog.map:    30,875
Loading eggnog.tre:    30,986
Loading interpro2go.map:    13,501
Loading interpro2go.tre:    29,204
Loading seed.map:    13,662
Loading seed.tre:    21,085
Meganizing: /data/home/leozenon/db/3.INF.daa
Meganizing init
Annotating DAA file using FAST mode (accession database and first accession per line)
Annotating references
10% 20% 30% 40% 50% 60% 70% 80% 90% 100% (17.0s)
Writing
10% 20% 30% 40% 50% 60% 70% 80% 90% 100% (3.1s)
100% (0.0s)
Binning reads Initializing...
Initializing binning...
WARNING: Not an RMA6 file, will ignore paired read information
Using 'Naive LCA' algorithm for binning: Taxonomy
Using Best-Hit algorithm for binning: SEED
Using Best-Hit algorithm for binning: EGGNOG
Using Best-Hit algorithm for binning: INTERPRO2GO
Binning reads...
100% (1.1s)
Binning reads Analyzing alignments
10% 20% 30% 40% 50% 60% 70% 80% 90% 100% (363.1s)
Total reads:        8,394,155
With hits:           8,394,155 
Alignments:        171,438,213
Assig. Taxonomy:     8,367,088
Assig. SEED:           300,729
Assig. EGGNOG:       4,889,350
Assig. INTERPRO2GO:  6,517,782
MinSupport set to: 4197
100% (0.0s)
Binning reads Applying min-support &amp; disabled filter to Taxonomy...
10% 20% 30% 40% 50% 60% 70% 80% 90% 100% (0.4s)
Min-supp. changes:       1,657
100% (21.3s)
Binning reads Writing classification tables
10% 20% 30% 40% 50% 60% 70% 80% 90% 100% (11.9s)
Binning reads Syncing
100% (0.1s)
Class. Taxonomy:           154
Class. SEED:               966
Class. EGGNOG:           4,235
Class. INTERPRO2GO:      8,936
Total time:  427s
Peak memory: 22.8 of 536.9G
</code></pre>
<h3 id="part-2---compute-comparison">Part 2 - Compute Comparison</h3>
<blockquote>
<p>As stated on the beginning of this section, this computation was only possible to be done on the Bioinformatics3 server due to MEGAN6 version.</p>
</blockquote>
<p>After <em>meganizing</em> all the <em>.daa</em> files the computation of all the comparisons is done using the command <code>xvfb-run ./tools/compute-comparison -i files -o output</code>.</p>
<p>Example:</p>
<pre><code>xvfb-run ./tools/compute-comparison -i db/mega/1.INF.daa db/mega/2.INF.daa db/mega/3.INF.daa db/mega/4.INF.daa db/mega/5.INF.daa db/mega/6.INF.daa db/mega/7.INF.daa db/mega/8.INF.daa db/mega/9.INF.daa db/mega/10.INF.daa db/mega/11.INF.daa db/mega/12.INF.daa db/mega/13.INF.daa db/mega/14.INF.daa db/mega/15.INF.daa db/mega/16.INF.daa db/mega/17.INF.daa db/mega/18.INF.daa db/mega/19.INF.daa db/mega/3.NO.daa db/mega/4.NO.daa db/mega/5.NO.daa db/mega/6.NO.daa db/mega/7.NO.daa db/mega/8.NO.daa db/mega/9.NO.daa db/mega/10.NO.daa db/mega/11.NO.daa db/mega/12.NO.daa db/mega/13.NO.daa db/mega/14.NO.daa db/mega/15.NO.daa db/mega/16.NO.daa db/mega/17.NO.daa db/mega/18.NO.daa db/mega/19.NO.daa -o compute.all.megan
</code></pre>
<p>The expected output is:</p>
<pre><code>[leozenon@bioinformatica3 ~]$ xvfb-run ./tools/compute-comparison -i all_.daa_meganized_files -o compute.all.megan
ES2 Prism: Error - GLX extension is not supported
    GLX version 1.3 or higher is required
Version   MEGAN Community Edition (version 6.18.5, built 14 Feb 2020)
Copyright (C) 2019 Daniel H. Huson. This program comes with ABSOLUTELY NO WARRANTY.
GLib-GIO-Message: 23:38:00.404: Using the 'memory' GSettings backend.  Your settings will not be saved or shared with other applications.
Executing: open file='db/mega/1.INF.daa' readOnly=true;update;
Loading MEGAN File: 1.INF.daa
Info: Opened file '1.INF.daa' with 10,933,575 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/2.INF.daa' readOnly=true;update;
Loading MEGAN File: 2.INF.daa
Opened file 'db/mega/2.INF.daa' with 8,612,519 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/3.INF.daa' readOnly=true;update;
Loading MEGAN File: 3.INF.daa
Opened file 'db/mega/3.INF.daa' with 8,394,155 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/4.INF.daa' readOnly=true;update;
Loading MEGAN File: 4.INF.daa
Opened file 'db/mega/4.INF.daa' with 9,920,947 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/5.INF.daa' readOnly=true;update;
Loading MEGAN File: 5.INF.daa
Opened file 'db/mega/5.INF.daa' with 7,699,142 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/6.INF.daa' readOnly=true;update;
Loading MEGAN File: 6.INF.daa
Opened file 'db/mega/6.INF.daa' with 9,409,751 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/7.INF.daa' readOnly=true;update;
Loading MEGAN File: 7.INF.daa
Opened file 'db/mega/7.INF.daa' with 8,974,968 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/8.INF.daa' readOnly=true;update;
Loading MEGAN File: 8.INF.daa
Opened file 'db/mega/8.INF.daa' with 6,616,825 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/9.INF.daa' readOnly=true;update;
Loading MEGAN File: 9.INF.daa
Opened file 'db/mega/9.INF.daa' with 8,837,959 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/10.INF.daa' readOnly=true;update;
Loading MEGAN File: 10.INF.daa
Opened file 'db/mega/10.INF.daa' with 10,470,593 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/11.INF.daa' readOnly=true;update;
Loading MEGAN File: 11.INF.daa
Opened file 'db/mega/11.INF.daa' with 9,517,603 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/12.INF.daa' readOnly=true;update;
Loading MEGAN File: 12.INF.daa
Opened file 'db/mega/12.INF.daa' with 9,551,516 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/13.INF.daa' readOnly=true;update;
Loading MEGAN File: 13.INF.daa
Opened file 'db/mega/13.INF.daa' with 12,039,786 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/14.INF.daa' readOnly=true;update;
Loading MEGAN File: 14.INF.daa
Opened file 'db/mega/14.INF.daa' with 8,711,183 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/15.INF.daa' readOnly=true;update;
Loading MEGAN File: 15.INF.daa
Opened file 'db/mega/15.INF.daa' with 9,211,980 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/16.INF.daa' readOnly=true;update;
Loading MEGAN File: 16.INF.daa
Opened file 'db/mega/16.INF.daa' with 11,314,084 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/17.INF.daa' readOnly=true;update;
Loading MEGAN File: 17.INF.daa
Opened file 'db/mega/17.INF.daa' with 11,975,812 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/18.INF.daa' readOnly=true;update;
Loading MEGAN File: 18.INF.daa
Opened file 'db/mega/18.INF.daa' with 7,793,529 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/19.INF.daa' readOnly=true;update;
Loading MEGAN File: 19.INF.daa
Opened file 'db/mega/19.INF.daa' with 10,105,954 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/3.NO.daa' readOnly=true;update;
Loading MEGAN File: 3.NO.daa
Opened file 'db/mega/3.NO.daa' with 6,887,738 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/4.NO.daa' readOnly=true;update;
Loading MEGAN File: 4.NO.daa
Opened file 'db/mega/4.NO.daa' with 7,967,337 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/5.NO.daa' readOnly=true;update;
Loading MEGAN File: 5.NO.daa
Opened file 'db/mega/5.NO.daa' with 8,655,375 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/6.NO.daa' readOnly=true;update;
Loading MEGAN File: 6.NO.daa
Opened file 'db/mega/6.NO.daa' with 7,933,496 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/7.NO.daa' readOnly=true;update;
Loading MEGAN File: 7.NO.daa
Opened file 'db/mega/7.NO.daa' with 8,841,339 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/8.NO.daa' readOnly=true;update;
Loading MEGAN File: 8.NO.daa
Opened file 'db/mega/8.NO.daa' with 3,933,650 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/9.NO.daa' readOnly=true;update;
Loading MEGAN File: 9.NO.daa
Opened file 'db/mega/9.NO.daa' with 6,924,690 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/10.NO.daa' readOnly=true;update;
Loading MEGAN File: 10.NO.daa
Opened file 'db/mega/10.NO.daa' with 11,334,423 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/11.NO.daa' readOnly=true;update;
Loading MEGAN File: 11.NO.daa
Opened file 'db/mega/11.NO.daa' with 8,432,444 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/12.NO.daa' readOnly=true;update;
Loading MEGAN File: 12.NO.daa
Opened file 'db/mega/12.NO.daa' with 10,121,509 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/13.NO.daa' readOnly=true;update;
Loading MEGAN File: 13.NO.daa
Opened file 'db/mega/13.NO.daa' with 8,554,302 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/14.NO.daa' readOnly=true;update;
Loading MEGAN File: 14.NO.daa
Opened file 'db/mega/14.NO.daa' with 7,049,651 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/15.NO.daa' readOnly=true;update;
Loading MEGAN File: 15.NO.daa
Opened file 'db/mega/15.NO.daa' with 6,801,339 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/16.NO.daa' readOnly=true;update;
Loading MEGAN File: 16.NO.daa
Opened file 'db/mega/16.NO.daa' with 6,367,556 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/17.NO.daa' readOnly=true;update;
Loading MEGAN File: 17.NO.daa
Opened file 'db/mega/17.NO.daa' with 7,168,004 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/18.NO.daa' readOnly=true;update;
Loading MEGAN File: 18.NO.daa
Opened file 'db/mega/18.NO.daa' with 7,515,418 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Executing: open file='db/mega/19.NO.daa' readOnly=true;update;
Loading MEGAN File: 19.NO.daa
Opened file 'db/mega/19.NO.daa' with 8,460,638 reads
Executing: update;
updating viewer
Induced tree has 1 of 1 nodes
updating viewer
Induced tree has 1 of 1 nodes
Computing comparison:
Normalizing to: 0 reads per sample
Total assigned:   62,038,417 normalized
Saving to file: compute.all.megan
done
Total time:  17s
Peak memory: 0.3 of 536.9G
</code></pre>
<p>After the computation it will be saved to compute.all.megan that can be analysed on the MEGAN6 GUI.</p>
<h1 id="plotting">Plotting</h1>
<p>The 5th part of the pipeline is designed to elaborate plots to help evaluate all the data processed.</p>
<ol>
<li>HISAT2</li>
<li>Samtools</li>
<li>DIAMOND</li>
<li>MEGAN6</li>
<li><strong>Plots</strong></li>
</ol>
<div class="mermaid"><svg xmlns="http://www.w3.org/2000/svg" id="mermaid-svg-pG6TMBCzWHn9A1Eb" width="100%" style="max-width: 831.171875px;" viewBox="0 0 831.171875 228"><g transform="translate(-12, -12)"><g class="output"><g class="clusters"><g class="cluster" id="subGraph0" transform="translate(700.671875,126)" style="opacity: 1;"><rect width="269" height="212" x="-134.5" y="-106"></rect><g class="label"><g transform="translate(0,0)"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"></div></foreignObject></g></g><text x="0" y="-92" fill="black" stroke="none" id="mermaid-svg-pG6TMBCzWHn9A1EbText" style="text-anchor: middle;">Plots</text></g></g><g class="edgePaths"><g class="edgePath" style="opacity: 1;"><path class="path" d="M94.734375,126L119.734375,126L144.734375,126" marker-end="url(#arrowhead2295)" style="fill:none"></path><defs><marker id="arrowhead2295" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M228.859375,126L253.859375,126L278.859375,126" marker-end="url(#arrowhead2296)" style="fill:none"></path><defs><marker id="arrowhead2296" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M378.390625,126L403.390625,126L428.390625,126" marker-end="url(#arrowhead2297)" style="fill:none"></path><defs><marker id="arrowhead2297" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M516.171875,126L541.171875,126L566.171875,126L591.171875,126" marker-end="url(#arrowhead2298)" style="fill:none"></path><defs><marker id="arrowhead2298" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M621.203125,115.18832766904957L672.8515625,78L724.5,78" marker-end="url(#arrowhead2299)" style="fill:none"></path><defs><marker id="arrowhead2299" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g><g class="edgePath" style="opacity: 1;"><path class="path" d="M621.203125,136.81167233095042L672.8515625,174L725.984375,174" marker-end="url(#arrowhead2300)" style="fill:none"></path><defs><marker id="arrowhead2300" viewBox="0 0 10 10" refX="9" refY="5" markerUnits="strokeWidth" markerWidth="8" markerHeight="6" orient="auto"><path d="M 0 0 L 10 5 L 0 10 z" class="arrowheadPath" style="stroke-width: 1; stroke-dasharray: 1, 0;"></path></marker></defs></g></g><g class="edgeLabels"><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="" style="opacity: 1;"><g transform="translate(0,0)" class="label"><foreignObject width="0" height="0"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel"></span></div></foreignObject></g></g><g class="edgeLabel" transform="translate(672.8515625,78)" style="opacity: 1;"><g transform="translate(-26.6484375,-13)" class="label"><foreignObject width="53.296875" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel">ggplot2</span></div></foreignObject></g></g><g class="edgeLabel" transform="translate(672.8515625,174)" style="opacity: 1;"><g transform="translate(-26.6484375,-13)" class="label"><foreignObject width="53.296875" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;"><span class="edgeLabel">ggplot2</span></div></foreignObject></g></g></g><g class="nodes"><g class="node" id="H" transform="translate(606.1875,126)" style="opacity: 1;"><rect rx="0" ry="0" x="-15.015625" y="-23" width="30.03125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-5.015625,-13)"><foreignObject width="10.03125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">R</div></foreignObject></g></g></g><g class="node" id="I" transform="translate(767.3359375,78)" style="opacity: 1;"><rect rx="0" ry="0" x="-42.8359375" y="-23" width="85.671875" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-32.8359375,-13)"><foreignObject width="65.671875" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">box-plots</div></foreignObject></g></g></g><g class="node" id="J" transform="translate(767.3359375,174)" style="opacity: 1;"><rect rx="0" ry="0" x="-41.3515625" y="-23" width="82.703125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-31.3515625,-13)"><foreignObject width="62.703125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">bar-plots</div></foreignObject></g></g></g><g class="node" id="A" transform="translate(57.3671875,126)" style="opacity: 1;"><rect rx="0" ry="0" x="-37.3671875" y="-23" width="74.734375" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-27.3671875,-13)"><foreignObject width="54.734375" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">HISAT2</div></foreignObject></g></g></g><g class="node" id="B" transform="translate(186.796875,126)" style="opacity: 1;"><rect rx="0" ry="0" x="-42.0625" y="-23" width="84.125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-32.0625,-13)"><foreignObject width="64.125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">Samtools</div></foreignObject></g></g></g><g class="node" id="C" transform="translate(328.625,126)" style="opacity: 1;"><rect rx="0" ry="0" x="-49.765625" y="-23" width="99.53125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-39.765625,-13)"><foreignObject width="79.53125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">DIAMOND</div></foreignObject></g></g></g><g class="node" id="D" transform="translate(472.28125,126)" style="opacity: 1;"><rect rx="0" ry="0" x="-43.890625" y="-23" width="87.78125" height="46"></rect><g class="label" transform="translate(0,0)"><g transform="translate(-33.890625,-13)"><foreignObject width="67.78125" height="26"><div xmlns="http://www.w3.org/1999/xhtml" style="display: inline-block; white-space: nowrap;">MEGAN6</div></foreignObject></g></g></g></g></g></g></svg></div>
<blockquote>
<p>Written with <a href="https://stackedit.io/">StackEdit</a>.</p>
</blockquote>
<hr class="footnotes-sep">
<section class="footnotes">
<ol class="footnotes-list">
<li id="fn1" class="footnote-item"><p>HUSON, D. H. et al. MEGAN Community Edition - Interactive Exploration and Analysis of Large-Scale Microbiome Sequencing Data. <strong>PLOS Computational Biology</strong>, v. 12, n. 6, p. e1004957, 21 jun. 2016. <a href="#fnref1" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn2" class="footnote-item"><p>BUCHFINK, B.; XIE, C.; HUSON, D. Fast and sensitive protein alignment using DIAMOND. <strong>Nature Methods</strong>, v. 12, n. 1, p. 59, 60 (2015).** <a href="#fnref2" class="footnote-backref">↩︎</a></p>
</li>
<li id="fn3" class="footnote-item"><p>Núcleo de Processamento de Alto Desempenho - NPAD - UFRN <a href="#fnref3" class="footnote-backref">↩︎</a></p>
</li>
</ol>
</section>
</div>
</body>

</html>
